GameLogic/python_emulator.py

```diff
@@ -0,0 +1,531 @@
+        for pixel in range(self.WIDTH):
+            pygame.draw.rect(
+                self.screen,
+                self.BLOCK_COLOR,
+                (pixel, self.GND_AXIS, self.GRID_SIZE, self.GRID_SIZE),
+            )
+
+    # Takes whichever sprite is dino_array currently and it's location and draws to the board
+    def draw_dino(self):
+        for row in range(self.dino_height):
+            for col in range(self.dino_width):
+                if self.dino_array[row, col] == 1:
+                    pygame.draw.rect(
+                        self.screen,
+                        self.BLOCK_COLOR,
+                        (
+                            self.dino_x + col * self.GRID_SIZE,
+                            self.dino_y + row * self.GRID_SIZE,
+                            self.GRID_SIZE,
+                            self.GRID_SIZE,
+                        ),
+                    )
+
+    # Determines which dino is shown: running LR, jumping, ducking LR, and game over (TODO)
+    def determine_and_draw_dino(self):
+        self.curr_walk -= 1  # counting down
+        if self.is_ducking and self.curr_walk == 0:
+            self.curr_walk = self.UPDATE_WALK_FRAMES
+            if self.is_right_foot_down:
+                self.is_right_foot_down = False
+                self.dino_array = self.ducking_left_array
+            else:
+                self.is_right_foot_down = True
+                self.dino_array = self.ducking_right_array
+        elif self.is_jumping:
+            self.dino_array = self.jump_array
+        elif self.curr_walk == 0:
+            self.curr_walk = self.UPDATE_WALK_FRAMES
+            if self.is_right_foot_down:
+                self.is_right_foot_down = False
+                self.dino_array = self.running_left_array
+            else:
+                self.is_right_foot_down = True
+                self.dino_array = self.running_right_array
+        self.draw_dino()
+
+    # Draws all cactus obstacles in the array and moves them to the left afterwards
+    def draw_cactus(self):
+        for x_loc in self.cactus_locs:
+            for row in range(self.cactus_height):
+                for col in range(self.cactus_width):
+                    if self.cactus_array[row, col] == 1:
+                        pygame.draw.rect(
+                            self.screen,
+                            self.BLOCK_COLOR,
+                            (
+                                x_loc + col * self.GRID_SIZE,
+                                self.cactus_y + row * self.GRID_SIZE,
+                                self.GRID_SIZE,
+                                self.GRID_SIZE,
+                            ),
+                        )
+        # Moves all cacti still on the board towards the dino at the moving speed
+        self.cactus_locs = [
+            cactus - self.real_moving_speed
+            for cactus in self.cactus_locs
+            if cactus - self.real_moving_speed >= -self.cactus_width
+        ]
```

Comment:
Similarly, consider putting the part of code that moves cactus to a different function, as it is not really about drawing them.